pkg/injector/envoy_container.go

```diff
@@ -15,21 +15,46 @@ const (
 	envoyProxyConfigPath     = "/etc/envoy"
 )
 
-func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes) corev1.Container {
+func getUnixSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		RunAsUser: func() *int64 {
+			uid := constants.EnvoyUID
+			return &uid
+		}(),
+	}
+}
+
+func getWindowsSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		WindowsOptions: &corev1.WindowsSecurityContextOptions{
+			RunAsUserName: func() *string {
+				userName := "EnvoyUser"
```

Comment:
I thought we were using one of the well known SIDs so that we could do prefix matching?

Reply:
Yeah we do but there is the following caveat:

In Envoy [Dockerfile](https://github.com/envoyproxy/envoy/blob/57bfbda2a50d2f026cf64c82383fe3bebe1e0860/ci/Dockerfile-envoy-windows#L8) we create the EnvoyUser and add it to the `Network Configuration Operators` group so we can do match prefix with this group

Reply:
Define a constant for `EnvoyUser` in this file.